src/core/public/chrome/doc_title/doc_title_service.ts

```diff
@@ -0,0 +1,180 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, Observable, ReplaySubject } from 'rxjs';
+import { compact, flattenDeep, isString, isArray } from 'lodash';
+import { takeUntil } from 'rxjs/operators';
+
+interface StartDeps {
+  document: { title: string };
+}
+
+/**
+ * {@link ChromeDocTitle | APIs} for accessing and updating the document title.
+ *
+ * @example
+ * How to change the title of the document
+ * ```ts
+ * chrome.docTitle.change({ parts: ['My application'], excludeBase: true })
+ * ```
+ *
+ * @example
+ * How to reset the title of the document to it's initial value
+ * ```ts
+ * chrome.docTitle.reset()
+ * ```
+ *
+ * @public
+ * */
+export interface ChromeDocTitle {
+  /**
+   * Gets an observable of the current document title.
+   */
+  get$: () => Observable<string>;
+  /**
+   * Changes the current document title.
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @example
+   * How to change the title of the document
+   * ```ts
+   * chrome.docTitle.change('My application title')
+   * chrome.docTitle.change(['My application', 'My section'])
+   * chrome.docTitle.change({ parts: ['My application'], excludeBase: true })
+   * ```
+   *
+   * @param newTitle The new title to set, either a string, string array or {@link ChromeDocTitleEntry}
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  change: (newTitle: ChromeDocTitleChange, apply?: boolean) => void;
+  /**
+   * Resets the document title to it's initial value.
+   * (meaning the one present in the title meta at application load.)
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  reset: (apply?: boolean) => void;
+  /**
+   * Apply a title change or reset that was not applied yet.
+   *
+   * @see {DocTitle#change}
+   * @see {DocTitle#reset}
+   */
+  apply: () => void;
+  /** @internal */
+  __legacy: {
+    setBaseTitle: (baseTitle: string) => void;
+  };
+}
```

Comment:
We don't have any other services with that kind of legacy API, however this was needed to keep the old docTitle karma test working, as they were using 'private' APIs to manually change to baseTitle of the old module.  Please tell me if these is a better approach

Reply:
What I've been doing with the legacy tests is just making them dumb unit tests that ensure the NP API was called by using a mock. That way we don't have to do things like this, which should save us time when we start removing the legacy code.

Reply:
@joshdover issue I had here is with the tests in `src/legacy/ui/public/doc_title/__tests__/doc_title.js` that were basically injected in the old docTitle module a different base name. Could find another way than than to keep the test. Do you see another approach ? Or maybe should I just remove the legacy test as we now got that covered in the NP module tests instead ?

Reply:
I'm fine with removing this test completely.